packages/collector/src/immediate.js

```diff
@@ -27,22 +26,14 @@ const agentOpts = require('./agent/opts');
 
 const isExcludedFromInstrumentation = coreUtil.excludedFromInstrumentation && coreUtil.excludedFromInstrumentation();
 
-// In case this is a child process of an instrumented parent process we might receive the agent uuid from the parent
-// process to be able to produce and collect spans immediately without waiting for a connection to the agent in this
-// process.
-// TODO: This does not work because you would report spans with parent agent uuid and the child process pid -
-//       this is not compatible. Our codebase does not support this.
-const parentProcessAgentUuid = process.env.INSTANA_AGENT_UUID;
+// CASE: This process is a forked child process of a bull worker.
+const currentProcessIsBullChildProcess = process.env.INSTANA_IS_BULL_CHILD_PROCESS === 'true';
```

Comment:
Changed from `INSTANA_AGENT_UUID` to exact env `INSTANA_IS_BULL_CHILD_PROCESS`.

This env is not documented INSTANA_AGENT_UUID.